lib/metasploit/framework/data_service/proxy/core.rb

```diff
@@ -52,30 +52,36 @@ def active
   end
 
   #
-  # Registers a data service with the proxy and immediately
-  # set as primary if online
+  # Registers the specified data service with the proxy
+  # and immediately sets it as the primary if active
   #
-  def register_data_service(data_service, online=false)
+  def register_data_service(data_service)
     validate(data_service)
     data_service_id = @data_service_id += 1
     @data_services[data_service_id] = data_service
-    set_data_service(data_service_id, online)
+    set_data_service(data_service_id)
   end
 
   #
   # Set the data service to be used
   #
-  def set_data_service(data_service_id, online=false)
+  def set_data_service(data_service_id)
     data_service = @data_services[data_service_id.to_i]
     if data_service.nil?
       raise "Data service with id: #{data_service_id} does not exist"
     end
 
-    if !online && !data_service.active
-      raise "Data service not online: #{data_service.name}, not setting as active"
+    if !data_service.is_local? && !data_service.active
```

Comment:
I usually prefer `unless` instead of negative checks, but I'm not sure what the style guide says.

Reply:
I’ve been using `unless` in situations that only involve a single variable and `if !` where more than one variable is involved for readability.